File: src/main/java/org/mifos/fineract/services/CashiersApi.java
package org.mifos.fineract.services;

import retrofit2.http.GET;
import retrofit2.http.Headers;
import rx.Observable;

public interface CashiersApi {
    /**
     * @param officeId (optional)
     * @param tellerId (optional)
     * @param staffId  (optional)
     * @param date     (optional)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("cashiers")
    Observable<String> getCashierData(
            @retrofit2.http.Query("officeId") Long officeId, @retrofit2.http.Query("tellerId") Long tellerId, @retrofit2.http.Query("staffId") Long staffId, @retrofit2.http.Query("date") String date
    );

}

File: src/main/kotlin/org/openapitools/client/apis/CashiersApi.kt
package org.openapitools.client.apis

import retrofit2.http.GET
import retrofit2.http.Query


interface CashiersApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param officeId  (optional)
     * @param tellerId  (optional)
     * @param staffId  (optional)
     * @param date  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/cashiers")
    suspend fun getCashierData(
        @Query("officeId") officeId: Long? = null,
        @Query("tellerId") tellerId: Long? = null,
        @Query("staffId") staffId: Long? = null,
        @Query("date") date: String? = null
    ): String

}
